mne/decoding/transformer.py

```diff
@@ -661,13 +676,28 @@ def fit(self, X, y=None):
         self : instance of UnsupervisedSpatialFilter
             Return the modified instance.
         """
+        # sklearn.utils.estimator_checks.check_estimator(self.estimator) is probably
+        # too strict for us, given that we don't fully adhere yet, so just check attrs
+        for attr in ("fit", "transform", "fit_transform"):
+            if not hasattr(self.estimator, attr):
+                raise ValueError(
+                    "estimator must be a scikit-learn "
+                    f"transformer, missing {attr} method"
+                )
+        _validate_type(self.average, bool, "average")
+        X = self._check_data(X, y=y, fit=True)
         if self.average:
             X = np.mean(X, axis=0).T
         else:
             n_epochs, n_channels, n_times = X.shape
             # trial as time samples
             X = np.transpose(X, (1, 0, 2)).reshape((n_channels, n_epochs * n_times)).T
+
+        # Really shouldn't modify self.estimator, but it's a bad backward compat problem
+        # to change it (will break people's pipelines) so leave it for now and just
+        # wrap to self.estimator_ as well
```

Comment:
should this come with a TODO (and/or get added to the roadmap), and get fixed later with a deprecation cycle?

Reply:
After thinking about it more I think it qualifies as a bugfix, pushed

Reply:
comment no longer applies.

```suggestion
```